File: wisefy/src/androidTest/java/com/isupatches/wisefy/main/IsDeviceConnectedToWifiNetworkTests.java
package com.isupatches.wisefy.main;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import com.isupatches.wisefy.AbstractBaseAndroidJUnit4TestClass;
import com.isupatches.wisefy.WiseFy;

import org.junit.Test;

/**
 * Tests the ability to determine if a device is connected to a wifi network.
 *
 * @see WiseFy#isDeviceConnectedToWifiNetwork()
 *
 * @author Patches
 */
public class IsDeviceConnectedToWifiNetworkTests extends AbstractBaseAndroidJUnit4TestClass {

  public IsDeviceConnectedToWifiNetworkTests() {
    // No-op
  }

  @Test
  public void failure_missingPrerequisite() {
    getMockWiseFyPrechecksUtil().isDeviceConnectedToWifiNetwork_failure();
    assertFalse(getWiseFy().isDeviceConnectedToWifiNetwork());
  }

  @Test
  public void failure() {
    getMockWiseFyConnectionUtil().isNetworkConnectedAndMatchesType(false);
    assertFalse(getWiseFy().isDeviceConnectedToWifiNetwork());
  }

  @Test
  public void success() {
    getMockWiseFyPrerequisitesUtil().activeNetwork();
    getMockWiseFyConnectionUtil().isNetworkConnectedAndMatchesType(true);
    assertTrue(getWiseFy().isDeviceConnectedToWifiNetwork());
  }
}

File: wisefy/src/androidTest/java/com/isupatches/wisefy/main/IsDeviceConnectedToWifiNetworkTests.kt
package com.isupatches.wisefy.main

import com.isupatches.wisefy.BaseAndroidJUnit4TestClass
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

/**
 * Tests the ability to determine if a device is connected to a wifi network.
 *
 * @author Patches
 */
internal class IsDeviceConnectedToWifiNetworkTests : BaseAndroidJUnit4TestClass() {

    @Test fun failure_missingPrerequisite() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToWifiNetwork_failure()
        assertFalse(wisefy.isDeviceConnectedToWifiNetwork())
    }

    @Test fun failure() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToWifiNetwork_success()
        mockWiseFyConnectionUtil.isDeviceConnectedToWifiNetwork(false)
        assertFalse(wisefy.isDeviceConnectedToWifiNetwork())
    }

    @Test fun success() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToWifiNetwork_success()
        mockNetworkUtil.activeNetwork()
        mockWiseFyConnectionUtil.isDeviceConnectedToWifiNetwork(true)
        assertTrue(wisefy.isDeviceConnectedToWifiNetwork())
    }
}
